test/actions/formThreats.js

```diff
@@ -0,0 +1,352 @@
+            response.data[0].location.should.be.equal('some_unq_location')
+          })
+        })
+      })
+    })
+
+    setup.describeAsUser(function (runAction) {
+      it('filter from_date', function () {
+        return runAction('formThreats:list', {from_date: '2016-12-20T10:15Z'}).then(function (response) {
+          response.should.not.have.property('error')
+          response.data.length.should.be.equal(2)
+        })
+      })
+      it('filter to_date', function () {
+        return runAction('formThreats:list', {to_date: '2016-12-20T10:16Z'}).then(function (response) {
+          response.should.not.have.property('error')
+          response.data.length.should.be.within(2, 4)// 3 records from fixtures
+        })
+      })
+      it('filter from_date and to_date', function () {
+        return runAction('formThreats:list', {from_date: '2016-12-20T10:15Z', to_date: '2016-12-20T10:16Z'}).then(function (response) {
+          response.should.not.have.property('error')
+          response.data.length.should.be.equal(1)
+        })
+      })
+    })
+  }) // given some threats rows
+
+  describe('Edit threats row', function () {
+    var threatsId
+
+    before(function () {
+      return setup.runActionAsUser2('formThreats:create', threatsRecord).then(function (response) {
+        threatsId = response.data.id
+      })
+    })
+
+    it('is allowed if the requester is the submitter', function () {
+      return setup.runActionAsUser2('formThreats:edit', {id: threatsId, notes: 'some new notes'}).then(function (response) {
+        response.should.not.have.property('error')
+        return setup.api.models.formThreats.findOne({where: {id: threatsId}}).then(function (threat) {
+          threat.notes.should.be.equal('some new notes')
+        })
+      })
+    })
+
+    setup.describeAsGuest(function (runAction) {
+      it('is not allowed if the requester is guest  user', function () {
+        return runAction('formThreats:edit', {id: threatsId}).then(function (response) {
+          response.should.have.property('error').and.not.empty()
+        })
+      })
+    })
+
+    setup.describeAsUser(function (runAction) {
+      it('is not allowed if the requester user is not the submitter', function () {
+        return runAction('formThreats:edit', {id: threatsId}).then(function (response) {
+          response.should.have.property('error').and.not.empty()
+        })
+      })
+    })
+
+    setup.describeAsAdmin(function (runAction) {
+      it('is allowed if the requester user is admin', function () {
+        return runAction('formThreats:edit', {id: threatsId, notes: 'some new notes'}).then(function (response) {
+          response.should.not.have.property('error')
+          return setup.api.models.formThreats.findOne({where: {id: threatsId}}).then(function (threat) {
+            threat.notes.should.be.equal('some new notes')
+          })
+        })
+      })
+    })
+  }) // Edit threats row
+})
```

Comment:
There should be tests for the specific required fields based on the selected types